File: demo/04-klassen/zylinder.py
from kreis import Kreis

class Zylinder(Kreis):
    
    def __init__(self, radius, hoehe):
        super().__init__(radius)
        self.hoehe = hoehe

    def volumen(self):
        return self.getFlaeche() * self.hoehe 

    def print(self):
        print("Zylinder mit Radius", self.radius, " und Höhe", self.hoehe)    
        print("- Volumen = ", self.volumen())


k = Kreis(5)
k.print()

z = Zylinder(5, 10)
z.print()

File: src/chapter08/inheritance/Zylinder.java
package chapter08.inheritance;

public class Zylinder extends Kreis {
	private double height;
	
	public Zylinder() {
		super();
	}
	
	public Zylinder(double radius, double height) {
		super(radius);
		this.height = height;
	}

	@Override
	public String toString() {
		return "Zylinder mit radius " + this.getRadius() 
				+ " und höhe " + this.getHeight()
				+ ": Volumen=" + getVolumen();
	}
	
	public double getVolumen() {
		return this.height * super.getFlaeche();
	}
	
	public double getHeight() {
		return height;
	}

	public void setHeight(double height) {
		this.height = height;
	}
	
}
